File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        if (root == nullptr) return {};
        queue<TreeNode *> q;
        q.push(root);
        vector<vector<int>> res;
        while (!q.empty()) {
            vector<int> temp;
            int size = q.size();
            for ( auto i = 0; i < size; i++ ) {
                auto node = q.front();
                q.pop();
                temp.emplace_back(node->val);
                if (node->left != nullptr) q.push(node->left);
                if (node->right != nullptr) q.push(node->right);
            }
            res.push_back(temp);
        }
        return res;
    }
};

File: Data Structures & Algorithms/quickSort/submission-0.py
# Definition for a pair.
# class Pair:
#     def __init__(self, key: int, value: str):
#         self.key = key
#         self.value = value
class Solution:
    def quickSort(self, pairs: List[Pair]) -> List[Pair]:
        return self.qucksortdfs(pairs,0,len(pairs)-1)
    def qucksortdfs(self,pairs:List[Pair],s,e):
        if e-s+1<=1:
            return pairs
        p = pairs[e]
        left = s
        #小的插入前面
        for i in range(s,e):
            if pairs[i].key < p.key:
                temp = pairs[left]
                pairs[left] = pairs[i]
                pairs[i] = temp
                left+=1
        pairs[e] = pairs[left]
        pairs[left] = p
        self.qucksortdfs(pairs,s,left-1)
        self.qucksortdfs(pairs,left+1,e)
        return pairs